astropy/cosmology/units.py

```diff
@@ -70,36 +70,50 @@ def dimensionless_redshift():
 def redshift_distance(cosmology=None, kind="comoving", **atzkw):
     """Convert quantities between redshift and distance.
 
-    Care should be taken to not misinterpret a relativistic, gravitational, etc
-    redshift as a cosmological one.
+    Care should be taken to not misinterpret a relativistic, gravitational, etc redshift
+    as a cosmological one.
 
     Parameters
     ----------
     cosmology : `~astropy.cosmology.Cosmology`, str, or None, optional
-        A cosmology realization or built-in cosmology's name (e.g. 'Planck18').
-        If None, will use the default cosmology
-        (controlled by :class:`~astropy.cosmology.default_cosmology`).
+        A cosmology realization or built-in cosmology's name (e.g. 'Planck18'). If None,
+        will use the default cosmology (controlled by
+        :class:`~astropy.cosmology.default_cosmology`).
     kind : {'comoving', 'lookback', 'luminosity'}, optional
-        The distance type for the Equivalency.
-        Note this does NOT include the angular diameter distance as this
-        distance measure is not monotonic.
+        The distance type for the Equivalency. Note this does NOT include the angular
+        diameter distance as this distance measure is not monotonic.
     **atzkw
-        keyword arguments for :func:`~astropy.cosmology.z_at_value`
+        keyword arguments for :func:`~astropy.cosmology.z_at_value`, which is used to
+        convert distance to redshift.
 
     Returns
     -------
     `~astropy.units.equivalencies.Equivalency`
         Equivalency between redshift and temperature.
 
+    Raises
+    ------
+    Exception
+        See :func:`~astropy.cosmology.z_at_value` for possible exceptions, e.g. if the
+        distance maps to a redshift that is larger than ``zmax``, the maximum redshift.
```

Comment:
The [NumPy docstring standard](https://numpydoc.readthedocs.io/en/latest/format.html#raises) says that the `Raises` section should describe only errors that are non-obvious or have a large chance of getting raised. Is the number of such errors very large? If not then it would be better to explicitly list them all. I don't like seeing `raises Exception`, not even in a docstring.

Reply:
Agreed about raising a general Exception, but this is just to sign post to users to check out `astropy.cosmology.z_at_value` for which this equivalency is a thin wrapper. That function can raise errors in a variety of ways, not least of which are the myriad errors raised by scipy, for which `z_at_value` is a wrapper.

Reply:
I can break out a specific `ValueError` for `zmax`.

Reply:
Done.